### PR TITLE
Enable Hardsigmoid for QNN EP using SDK support direct support

### PR DESCRIPTION
**Pull request by HectorSVC**
### Description
Enable Hardsigmoid for QNN EP using SDK support direct support instead of decomposing to its constituent ops so it can support the quantized model

### CODE REVIEW

File: onnxruntime/test/providers/qnn/simple_op_htp_test.cc [hunk 1]
```diff
@@ -1246,8 +1258,15 @@ TEST_F(QnnHTPBackendTests, UnaryOp_HardSigmoid_F32_as_FP16) {
...
-TEST_F(QnnHTPBackendTests, UnaryOp_HardSigmoid_FP16) {
+// Check that QNN EP can support float16 HardSigmoid on HTP
+// \HTP\src\hexagon\prepare\tcm_migration.cc:1829:ERROR:no properties registered for q::QNN_HardSigmoid
+// \HTP\HTP\src\hexagon\prepare\graph_prepare.cc:203:ERROR:could not create op: q::QNN_HardSigmoid
+// \HTP\HTP\src\hexagon\prepare\graph_prepare.cc:1238:ERROR:Op 0x101000000010 preparation failed with err:-1
+// Completed stage: Graph Transformations and Optimizations (16361 us)
+// QnnDsp <E> "node" generated: could not create op
+// QnnDsp <E> RouterWindows graph prepare failed 12
+// QnnDsp <E> Failed to finalize graph (id: 1) with err 1002
+TEST_F(QnnHTPBackendTests, DISABLED_UnaryOp_HardSigmoid_FP16) {
```
Comment (adrianlizarraga) [inline onnxruntime/test/providers/qnn/simple_op_htp_test.cc:1269]: I think we should consider keeping the decomposition only for the FP16 HardSigmoid on HTP. FP16 HardSigmoid currently works, but now we're removing it.

Comment (HectorSVC) [reply onnxruntime/test/providers/qnn/simple_op_htp_test.cc:1269]: This is a bug in QNN. The problem is how to provide QC a reproduce. I think for most of the case, user just use the fp32 model with FP16 precision enabled. It works with that approach. This only happens to the converted FP16 model.

Comment (adrianlizarraga) [reply onnxruntime/test/providers/qnn/simple_op_htp_test.cc:1269]: Right, this is a bug in QNN. However, ort 1.18 shipped with the ability for the user to run fp16 HardSigmoid on HTP. There could be some users that are currently relying on this now, and this PR would take away that capability unnecessarily. We have the ability to still support fp16 HardSigmoid until the bug is resolved by QC.

Comment (HectorSVC) [reply onnxruntime/test/providers/qnn/simple_op_htp_test.cc:1269]: make sense

Comment (adrianlizarraga) [reply onnxruntime/test/providers/qnn/simple_op_htp_test.cc]: Thank you!

**Review by adrianlizarraga — APPROVED** [review]

**Pull request merged**